recipes/tree-sitter-c/all/conanfile.py

```diff
@@ -13,60 +15,48 @@ class TreeSitterCConan(ConanFile):
     homepage = "https://github.com/tree-sitter/tree-sitter-c"
     license = "MIT"
     settings = "os", "arch", "compiler", "build_type"
+    package_type = "shared-library"
```

Comment:
Can you provide a bit more context about this? Why removing the shared option and setting to shared-library? Is it needed?

Reply:
My understanding is that Tree-sitter is a GLR parser that implements the main parsing algorithms. Therefore, when used as a library, it can be either statically or dynamically linked. Tree-sitter provides different language grammars as separate libraries, such as tree-sitter-c, tree-sitter-cpp, and tree-sitter-json. These libraries can be loaded on-demand, functioning as dynamic link libraries.

In fact, the Tree-sitter team maintains Python bindings, and they also build these language libraries as dynamic link libraries. You can refer to their python binding for tree-sitter-[x]: [py-tree-sitter](https://github.com/tree-sitter/py-tree-sitter/blob/d3016edac2c33ce647653d896fbfb435ac2a6245/tree_sitter/__init__.py#L17-L74).

However, for this library, both static linking and dynamic linking are ok.

Reply:
In fact, in most cases that I have seen, language grammars are presented as dynamic link libraries. For example, this is the case with packages in the Arch User Repository (AUR) and those used by Neovim.

Reply:
> Can you provide a bit more context about this? Why removing the shared option and setting to shared-library? Is it needed?

Hey, @czoido Is there any other stuffs unclear to you? don't hesitate to ask me anything.